qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -0,0 +1,810 @@
+            limit_amplitude: If ``True``, then limit the absolute value of the amplitude of the
+                waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+            envelope: Pulse envelope expression.
+            constraints: Pulse parameter constraint expression.
+            eval_conditions: Extra conditions to evaluate a full waveform to check the
+                amplitude limit. If this condition is met, then the validation routine
+                will investigate the full waveform and raise an error when the amplitude norm
+                of any data point exceeds 1.0. If not provided, the validation always
+                creates a full waveform.
+
+        Raises:
+            PulseError: When not all parameters are listed in the attribute :attr:`PARAM_DEF`.
+        """
+        super().__init__(
+            duration=duration,
+            name=name,
+            limit_amplitude=limit_amplitude,
+        )
+        if parameters is None:
+            parameters = {}
+
+        self._pulse_type = pulse_type
+        self._params = parameters
+
+        self.envelope = envelope
+        self.constraints = constraints
+        self.eval_conditions = eval_conditions
+
+    def __getattr__(self, item):
+        # Get pulse parameters with attribute-like access.
+        params = object.__getattribute__(self, "_params")
+        if item not in params:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{item}'")
+        return params[item]
+
+    @property
+    def pulse_type(self) -> str:
+        """Return display name of the pulse shape."""
+        return self._pulse_type
+
+    def get_waveform(self) -> Waveform:
+        r"""Return a Waveform with samples filled according to the formula that the pulse
+        represents and the parameter values it contains.
+
+        Since the returned array is a discretized time series of the continuous function,
+        this method uses a midpoint sampler. For ``duration``, return:
+
+        .. math::
+
+            \{f(t+0.5) \in \mathbb{C} | t \in \mathbb{Z} \wedge  0<=t<\texttt{duration}\}
+
+        Returns:
+            A waveform representation of this pulse.
+
+        Raises:
+            PulseError: When parameters are not assigned.
+            PulseError: When expression for pulse envelope is not assigned.
+        """
+        if self.is_parameterized():
+            raise PulseError("Unassigned parameter exists. All parameters must be assigned.")
+
+        if self.envelope is None:
+            raise PulseError("Pulse envelope expression is not assigned.")
+
+        fargs = _get_expression_args(self.envelope, self.parameters)
+        return Waveform(samples=self._envelope_lam(*fargs), name=self.name)
+
+    def validate_parameters(self) -> None:
+        """Validate parameters.
+
+        Raises:
+            PulseError: If the parameters passed are not valid.
```

Comment:
No parameters are `passed` to this function. Can you clarify?

Reply:
Thanks, added error message for this. https://github.com/Qiskit/qiskit-terra/pull/7821/commits/8edaba96a553f8cf3aef0b187bf18d26f5cf4455